tests/manage/monitoring/test_workload_with_distruptions.py

```diff
@@ -172,3 +178,168 @@ def test_workload_with_checksum_verify(
+        # otherwise the failure is legitimate
+        storageutilization_min_mbps = config.ENV_DATA[
+            "fio_storageutilization_min_mbps"
+        ] / len(get_osd_pods())
+        self.timeout_sec = get_timeout(storageutilization_min_mbps, int(pvc_size))
+
+        access_modes = [constants.ACCESS_MODE_RWO, constants.ACCESS_MODE_RWX]
+
+        num_of_load_objs = 2
+        self.pvc_objs = multi_pvc_factory(
+            interface=constants.CEPHFILESYSTEM,
+            size=pvc_size / num_of_load_objs,
+            access_modes=access_modes,
+            status=constants.STATUS_BOUND,
+            num_of_pvc=num_of_load_objs,
+            wait_each=True,
+        )
+        self.pod_objs = []
+
+        for pvc_obj in self.pvc_objs:
+            pod_obj = pod_factory(
+                interface=constants.CEPHFILESYSTEM, pvc=pvc_obj, replica_count=3
+            )
+            self.pod_objs.append(pod_obj)
+            file_name = pod_obj.name
+            pod_obj.fillup_fs(size=f"{round(pvc_size * 1024)}M", fio_filename=file_name)
+            pod_obj.run_io(
+                storage_type="fs",
+                size="3G",
+                runtime=self.timeout_sec,
+                fio_filename=f"{pod_obj.name}_io",
+            )
+
+        self.start_workload_time = time.perf_counter()
+
+        def finalizer():
+            """
+            Set default values for:
+              osd_op_complaint_time=30.000000
+            """
+            # set the osd_op_complaint_time to selected monitor back to default value
+            set_osd_op_complaint_time(constants.DEFAULT_OSD_OP_COMPLAINT_TIME)
+
+            # delete resources
+            for pod_obj in self.pod_objs:
+                pod_obj.delete()
+                pod_obj.delete(wait=True)
+
+            for pvc_obj in self.pvc_objs:
+                pvc_obj.delete(wait=True)
+
+        request.addfinalizer(finalizer)
+
+    @tier3
+    @blue_squad
+    def test_ceph_osd_low_ops_alert(self, setup):
+        """
+        Test to verify bz #1966139, more info about Prometheus alert - #1885441
+
+        CephOSDSlowOps. An Object Storage Device (OSD) with slow requests is every OSD that is not able to service
+        the I/O operations per second (IOPS) in the queue within the time defined by the osd_op_complaint_time
+        parameter. By default, this parameter is set to 30 seconds.
+
+        1. As precondition test setup is to reduce osd_op_complaint_time to 0.1 to prepare condition
+        to get CephOSDSlowOps
+        2. Run workload_fio_storageutilization gradually filling up the storage up to 90% in a background
+        2.1 Validate the CephOSDSlowOps fired if so check an alert message and finish the test
+        2.2 If CephOSDSlowOps has not been fired while the storage filled up to 90% or time to fill up the storage ends
+        - fail the test
+        """
+
+        api = PrometheusAPI()
+
+        while get_percent_used_capacity() < 89:
```

Comment:
It would be good to have a timeout for the function so that if something prevents the cluster from being utilized then it will fail after a while and doesn't run indefinitely.

Reply:
it has a timeout check 2 bellow. 
            if time_passed_sec > self.timeout_sec:
                pytest.fail("failed to fill the storage in calculated time")